src/scipp/testing/strategies.py

```diff
@@ -0,0 +1,225 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+# @author Jan-Lukas Wynen
+from functools import partial
+from typing import Optional, Sequence, Union
+
+from hypothesis.errors import InvalidArgument
+from hypothesis import strategies as st
+from hypothesis.extra import numpy as npst
+
+from ..core import variable as creation
+from ..core import DataArray, DType
+
+# TODO sometimes checks fail with (seen in test_astype_int_to_float)
+#  hypothesis.errors.FailedHealthCheck: Examples routinely exceeded the max
+#  allowable size. (20 examples overran while generating 9 valid ones).
+#  Generating examples this large will usually lead to bad results.
+#  You could try setting max_size parameters on your collections and turning
+#  max_leaves down on recursive() calls.
+#  See https://hypothesis.readthedocs.io/en/latest/healthchecks.html for more
+#  information about this. If you want to disable just this health check,
+#  add HealthCheck.data_too_large to the suppress_health_check settings for this test.
+#
+#  Maybe the variable strat has too many parameters or too many nested strats?
+#  Nested strats are reinited for every example,
+#  need to change to make them once up front
+#  But that is problematic because with_variances depends on dtype.
```

Comment:
For the purpose of strategies, pretend that `with_variances` is part of the dtype, so you can avoid the nesting?

Reply:
Not sure what you are suggesting. How would you encode both params via `dtype`?

Reply:
Instead of passing an actual scipp dtype pass, e.g., a string. You could have `float64, float64_with_error, int64, bool, ...` as options.